File: python/src/dogma/latitude.py
# This is free and unencumbered software released into the public domain.

"""The dogma.Latitude class."""

from .angle import Angle

# See: https://dogma.dev/Latitude
class Latitude(Angle):
    """A latitude."""

    MIN_DEGREES = -90
    MAX_DEGREES = 90

    def __init__(self, degrees: float) -> None:
        degrees = float(degrees)
        if abs(degrees) > Latitude.MAX_DEGREES:
            raise ValueError(f"Latitude bounds are ±90°, but got {degrees}°")
        super().__init__(degrees=degrees)

File: ruby/lib/dogma/latitude.rb
# This is free and unencumbered software released into the public domain.

require_relative 'angle'

##
# A latitude.
#
# @see https://dogma.dev/Latitude
class Dogma::Latitude < Dogma::Angle
  MIN_DEGREES = -90
  MAX_DEGREES = 90

  ##
  # @param  [Float, #to_f] degrees (-90..90)
  # @return [void]
  # @raise  [ArgumentError] if `degrees` is out of bounds
  def initialize(degrees)
    degrees = degrees.to_f
    raise ArgumentError, "Latitude bounds are ±90°, but got #{degrees}°" if degrees.abs > MAX_DEGREES
    super(degrees: degrees)
  end
end # Dogma::Latitude
